Neos.Neos/Classes/Controller/Backend/SchemaController.php

```diff
@@ -33,26 +37,52 @@ class SchemaController extends ActionController
      */
     protected $nodeTypeSchemaBuilder;
 
+    /**
+     * @Flow\Inject
+     * @var VariableFrontend
+     */
+    protected $nodeTypeSchemaCache;
+
     /**
      * Generate and renders the JSON schema for the node types for VIE.
      * Schema format example: http://schema.rdfs.org/all.json
      *
      * @return string
      */
-    public function vieSchemaAction()
+    public function vieSchemaAction(): string
     {
+        $version = $this->request->hasArgument('version') ? $this->request->getArgument('version') : '';
+        $cacheIdentifier = 'vieSchema_' . $version;
+
         $this->response->setContentType('application/json');
-        return json_encode($this->vieSchemaBuilder->generateVieSchema());
+        $this->response->setComponentParameter(SetHeaderComponent::class, 'Cache-Control', 'max-age=' . (3600 * 24 * 7));
+
+        $vieSchema = $this->nodeTypeSchemaCache->get($cacheIdentifier);
+        if (!$vieSchema) {
+            $vieSchema = json_encode($this->vieSchemaBuilder->generateVieSchema());
+            $this->nodeTypeSchemaCache->set($cacheIdentifier, $vieSchema);
```

Comment:
This would add an entry for every configuration change, which is inefficient and pollutes the cache, as we will never will reuse an old version.

Reply:
A cache flush could be somehow connected to configurationManager / FileMonitor ...